library/src/main/java/com/datatorrent/lib/io/block/ReaderContext.java

```diff
@@ -190,11 +190,11 @@ protected Entity readEntity() throws IOException
           if (bytesRead == -1) {
             break;
           }
-          strBuffer = new String(buffer);
+          bufferStr = new String(buffer,0, bytesRead);
         }
 
-        while (posInStr < strBuffer.length()) {
-          char c = strBuffer.charAt(posInStr);
+        while (posInStr < bufferStr.length()) {
+          char c = bufferStr.charAt(posInStr);
           if (c != '\r' && c != '\n') {
             tmpBuilder.write(c);
```

Comment:
@yogidevendra What does it take to address the tmpBuilder issue now? Does not seem to require elaborate discussion. I would also prefer to get it fixed but if you are not able to work on it then take it up for next release.